src/ImageSearchModal.jsx

```diff
@@ -9,244 +9,214 @@ import { Radio } from "@patternfly/react-core/dist/esm/components/Radio";
+                    </FormGroup>
+                    <FormGroup fieldId="registry-select" label={_("in")}>
+                        <FormSelect id='registry-select'
+                            value={selectedRegistry}
+                            onChange={(_ev, value) => { setSelectedRegistry(value); onSearchTriggered(value, false) }}>
+                            <FormSelectOption value="" key="all" label={_("All registries")} />
+                            {(registries.search || []).map(r => <FormSelectOption value={r} key={r} label={r} />)}
+                        </FormSelect>
+                    </FormGroup>
+                </Flex>
+            </Form>
+
+            {searchInProgress && <EmptyStatePanel loading title={_("Searching...")} /> }
+
+            {((!searchInProgress && !searchFinished) || imageIdentifier == "") && <EmptyStatePanel title={_("No images found")} paragraph={_("Start typing to look for images.")} /> }
+
+            {searchFinished && imageIdentifier !== '' && <>
+                {imageList.length == 0 && <EmptyStatePanel icon={ExclamationCircleIcon}
+                                                                      title={cockpit.format(_("No results for $0"), imageIdentifier)}
+                                                                      paragraph={_("Retry another term.")}
+                />}
+                {imageList.length > 0 &&
+                <DataList isCompact
+                          selectedDataListItemId={"image-list-item-" + selected}
+                          onSelectDataListItem={(_, key) => setSelected(key.split('-').slice(-1)[0])}>
+                    {imageList.map((image, iter) => {
+                        return (
+                            <DataListItem id={"image-list-item-" + iter} key={iter}>
+                                <DataListItemRow>
+                                    <DataListItemCells
+                                              dataListCells={[
+                                                  <DataListCell key="primary content">
+                                                      <span className='image-name'>{image.Name}</span>
+                                                  </DataListCell>,
+                                                  <DataListCell key="secondary content">
+                                                      <span className='image-description'>{image.Description}</span>
+                                                  </DataListCell>
+                                              ]}
+                                    />
+                                </DataListItemRow>
+                            </DataListItem>
+                        );
+                    })}
+                </DataList>}
+            </>}
+        </>
+    );
+
+    const handleClose = () => {
+        if (activeConnection)
+            activeConnection.close();
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1332-20230704-165425-9756b240-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)